src/app/recommendations.ts

```diff
@@ -13,210 +19,230 @@ export interface Step {
-  { possibleIn: 1400, necessaryAsOf: 1400, level: 3, step: 'v14 LocationStrategy', action: 'Implementing the `LocationStrategy` interface now requires definition of `getState()`.' },
-  { possibleIn: 1400, necessaryAsOf: 1400, level: 3, step: 'v14 http queries', action: 'Sending `+` as part of a query no longer requires workarounds since `+` no longer sends a space.' },
-  { possibleIn: 1400, necessaryAsOf: 1400, level: 3, step: 'v14 AnimationDriver.getParentElement', action: 'Implementing `AnimationDriver` now requires the `getParentElement` method.' },
-  { possibleIn: 1400, necessaryAsOf: 1400, level: 3, step: 'v14 invalid config', action: 'Invalid route configurations of lazy-loaded modules will now throw an error rather than being ignored.' },
-  { possibleIn: 1400, necessaryAsOf: 1400, level: 3, step: 'v14 router resolver', action: 'Remove the `resolver` from `RouterOutletContract.activateWith` function and the `resolver` from `OutletContext` class since factory resolvers are no longer needed.' },
-  { possibleIn: 1400, necessaryAsOf: 1400, level: 3, step: 'v14 initialUrl', action: '`Router.initialUrl` accepts only `UrlTree` to prevent a misuse of the API by assigning a `string` value.' },
+  { possibleIn: 1500, necessaryAsOf: 1500, level: ApplicationComplexity.Basic, step: 'v15 node support', action: 'Angular v15 supports node.js versions: 14.20.x, 16.13.x and 18.10.x. Make sure that you are using a supported version before you upgrade Angular. <a href="https://angular.io/guide/update-to-version-15#v15-bc-01" alt="Link to more information about this change">Read further</a>' },
+  { possibleIn: 1500, necessaryAsOf: 1500, level: ApplicationComplexity.Basic, step: 'v15 ts support', action: 'Angular v15 supports TypeScript version 4.8 or later.  Make sure that you are using a supported version before you upgrade Angular. <a href="https://angular.io/guide/update-to-version-15#v15-bc-02" alt="Link to more information about this change">Read further</a>' },
+  { possibleIn: 1500, necessaryAsOf: 1500, level: ApplicationComplexity.Basic, step: 'v15 keyframe', action: 'In v15, `@keyframes` names are prefixed with the component\'s <emphasis>scope name</emphasis>. <a href="https://angular.io/guide/update-to-version-15#v15-bc-03" alt="Link to more information about this change">Read further</a>' },
+  { possibleIn: 1500, necessaryAsOf: 1500, level: ApplicationComplexity.Basic, step: 'v15 no-ivey', action: 'In v15, the Angular compiler option `enableIvy` has been removed' },
+  { possibleIn: 1500, necessaryAsOf: 1500, level: ApplicationComplexity.Basic, step: '', action: 'In v15, when a class inherits its constructor from a base class, the compiler can report an error when that constructor cannot be used for dependency injection purposes. <a href="https://angular.io/guide/update-to-version-15#v15-bc-05" alt="Link to more information about this change">Read further</a>'},
```

Comment:
```suggestion
  { possibleIn: 1500, necessaryAsOf: 1500, level: ApplicationComplexity.Medium, step: '', action: 'Make sure you use decorators in base classes when their child classes inherit constructors and use dependency injection. Base classes should be decorated with `@Injectable` or `@Directive` alternatively the compiler will throw an error. <a href="https://angular.io/guide/update-to-version-15#v15-bc-05" alt="Link to more information about this change">Read further</a>'},
```

Reply:
for:
`Base classes should be decorated with `@Injectable` or `@Directive` alternatively the compiler will throw an error.`
would this work?
`Base classes should be decorated with either `@Injectable` or `@Directive` or the compiler returns an error.`

Reply:
Yes, this sounds good. `@Directive` should be uppercase, just like `@Injectable`.

Reply:
Updated in latest commit.